cache/refs.go

```diff
@@ -46,7 +48,7 @@ type ImmutableRef interface {
 	Finalize(context.Context) error
 
 	Extract(ctx context.Context, s session.Group) error // +progress
-	GetRemotes(ctx context.Context, createIfNeeded bool, compressionopt compression.Config, all bool, s session.Group) ([]*solver.Remote, error)
+	GetRemotes(ctx context.Context, createIfNeeded bool, cfg config.RefConfig, all bool, s session.Group) ([]*solver.Remote, error)
```

Comment:
@ktock @sipsma Any ideas how to improve this? Still not very clean. And a bit of an issue with `RefConfig` is that keys can be uninitialized (eg. `Compression` doesn't have a clear zero value atm). We can do this refactoring later in a follow-up PR.

Reply:
Maybe functional options?